### DISCUSSION TITLE
goreplay service concepts

### DISCUSSION THREAD

**Discussion by xingren23** [probelabs/goreplay#984] [Ideas]
GoReplay Service Concepts
1. GoReplay service can be registered as traffic recording role, replay role, or both in one. and provides the traffic recording and replay service interface.
2. The Admin is responsible for scheduling tasks to the corresponding service instances.
3. GoReplay instances provide real-time feedback on the status of tasks.

if so, you can use GoReplay services to automate traffic recording and replay.
Is it possible ?

**Comment by buger** [comment]
Hi! 

In general, I'm working in this direction and have similar views. 
Step number one to provide proper configuration management https://github.com/buger/goreplay/pull/874
This PR also gives ability to load config remotely, so next step will be dynamic config loading. 
Additional, this PR adds notion of Services, and you can have separate Services with different configs. 

Next step will be exposing some simple API which will allow to set and get config dynamically. 
Another step in this direction is built-in analytics. 

So it is in works.
